File: sdk/nodejs/cmd/pulumi-language-nodejs/testdata/published/tsnode/projects/l3-component-nested/outerComponent.ts
import * as pulumi from "@pulumi/pulumi";
import { InnerComponent } from "./innerComponent";

interface OuterComponentArgs {
    /**
     * An input passed to the outer component
     */
    input: pulumi.Input<boolean>,
}

export class OuterComponent extends pulumi.ComponentResource {
    public output: pulumi.Output<boolean>;
    constructor(name: string, args: OuterComponentArgs, opts?: pulumi.ComponentResourceOptions) {
        super("components:index:OuterComponent", name, args, opts);
        const innerComponent = new InnerComponent(`${name}-innerComponent`, {input: !args.input}, {
            parent: this,
        });

        this.output = innerComponent.output;
        this.registerOutputs({
            output: innerComponent.output,
        });
    }
}

File: sdk/python/cmd/pulumi-language-python/testdata/classes/local/projects/l3-component-nested/outerComponent.py
import pulumi
from innerComponent import InnerComponent
from pulumi import Input
from typing import Optional, Dict, TypedDict, Any
import builtins as _builtins

class OuterComponentArgs(TypedDict):
    input: Input[_builtins.bool]

class OuterComponent(pulumi.ComponentResource):
    def __init__(self, name: str, args: OuterComponentArgs, opts:Optional[pulumi.ResourceOptions] = None):
        super().__init__("components:index:OuterComponent", name, args, opts)

        inner_component = InnerComponent(f"{name}-innerComponent", {
            'input': not args["input"]},
            opts = pulumi.ResourceOptions(parent=self))

        self.output = inner_component.output
        self.register_outputs({
            'output': inner_component.output
        })